spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/support/AbstractRibbonCommand.java

```diff
@@ -114,8 +114,16 @@ protected ClientHttpResponse run() throws Exception {
 		final RequestContext context = RequestContext.getCurrentContext();
 
 		RQ request = createRequest();
-		RS response = this.client.executeWithLoadBalancer(request, config);
-
+		RS response;
+		// Note: using isRetriable to determine if spring-retry or netflix is are managing loadbalancing concerns.
```

Comment:
I know @ryanjbaxter  and @spencergibb have an outstanding concern around the 'isRetryable()' flag and in-line comments.
>Can you explain in a bit more detail about why isRetriable() says it's spring retry vs not? As far as I know, it's if you >are using ribbon RestClient or not.

@spencergibb and @ryanjbaxter this is a good question. There are 3 configurations I have found that affect the restclient configuration.

1 Having a RetryTemplate defined determines if a RetryableRibbonLoadBalancingHttpClient is used otherwise its a RibbonLoadBalancingHttpClient.
2 The zuul.retryable property is used as a flag in several locations for defining a request as retryable or not.
3 The ribbon.httpclient.enabled property enables the HttpClientRibbonConfiguration

looking at these three configurations I would say that the ribbon.httpclient.enabled property most likely defines the RestClient as ribbon or not. Is this correct?

overall its a challenging configuration, I will give some more thought to inline comments.

Thanks again

Reply:
`ribbon.restclient.enabled` is the only way to determine if `RestClient` is used.

Reply:
Thanks Spencer.. I am thinking the following single comment might be more appropriate.  would you agree?
> // using isRetriable flag to select either the LBC client or LoadBalancerCommand for executing the request and handling both retry and loadbalancing concerns.

Reply:
I guess I'm questioning that `isRetriable()` is the right thing to use here. I haven't seen any evidence that it is correct.

Reply:
I agree with your concern, both the AbstractLoadBalancerAwareClient implementation and the LoadBalancerCommand should be retriable provided they are configured to do so.  With that understanding using 'isRetriable()' to switch between them may not be valid.  I will take some time to review the code with a focus on the ribbon.restclient.enabled property in search for either solid evidence or a better solution.